crates/viewer/re_view_spatial/src/visualizers/transform_axes_3d.rs

```diff
@@ -125,21 +133,60 @@ impl VisualizerSystem for TransformAxes3DVisualizer {
                 None,
                 &latest_at_query,
                 data_result,
-                [TransformAxes3D::descriptor_axis_length().component],
+                [axis_length_identifier, show_frame_identifier],
                 false,
             );
 
             let axis_length: f32 = results
-                .get_mono_with_fallback::<AxisLength>(
-                    TransformAxes3D::descriptor_axis_length().component,
-                )
+                .get_mono_with_fallback::<AxisLength>(axis_length_identifier)
                 .into();
 
             if axis_length == 0.0 {
                 // Don't draw axis and don't add to the bounding box!
                 continue;
             }
 
+            let show_frame: bool = results
+                .get_mono_with_fallback::<ShowLabels>(show_frame_identifier)
+                .into();
+
+            if show_frame {
+                let frame_id_hash =
+                    transforms.transform_frame_id_for(data_result.entity_path.hash());
+
+                if let Some(frame_id) = transforms.lookup_frame_id(frame_id_hash) {
+                    // Add label at the center of each transform instance if show_frame is enabled
```

Comment:
I think it should be called `show_label`. Reading `show_frame` without such documentation (which anyway explains it as a text _label_), I would at first think that it's related to the coordinate frame markers or so.